### PR TITLE
set ref id length to 18

### PR DESCRIPTION
**Pull request by yuhanniu**
ref id will be mapped to fasterPayments reference information field which has a maximum length of 18 characters.

### CODE REVIEW

**Comment by zixunjian** [comment]
should we do this for access tokens too for consistency?

**Comment by zixunjian** [comment]
also what if `setRefId` is called with a string longer than 18?

**Comment by yuhanniu** [comment]
@zixunjian Good points, done

File: src/main/AccessTokenBuilder.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ export default class AccessTokenBuilder {
 
         this._payload = {
             version: config.accessTokenVersion,
-            refId: Util.generateNonce(),
+            refId: Util.generateNonce().substr(0, 18),
```
Comment (colin-token) [inline src/main/AccessTokenBuilder.js]: add optional length to generateNonce?

Comment (yuhanniu) [reply src/main/AccessTokenBuilder.js]: I think this is clean enough, your suggestion is more like an optimization

Comment (yuhanniu) [reply src/main/AccessTokenBuilder.js]: Will need to rewrite `generateNonce`, to generate given length of nonce

**Pull request merged**